src/main/java/io/quarkus/search/app/quarkusio/QuarkusIO.java

```diff
@@ -95,6 +100,39 @@ public QuarkusIO(QuarkusIOConfig config, GitCloneDirectory mainRepository,
         this.localizedSiteUris = localizedSites.entrySet().stream().collect(
                 Collectors.toUnmodifiableMap(Map.Entry::getKey, e -> config.localized().get(e.getKey().code).webUri()));
         this.failureCollector = failureCollector;
+
+        this.localizedSitesUpToDate = Collections.unmodifiableMap(
+                createUpToDateLocalizedSitesMap(mainRepository, localizedSites, failureCollector));
+    }
+
+    private static Map<Language, Boolean> createUpToDateLocalizedSitesMap(
+            GitCloneDirectory mainRepository,
+            Map<Language, GitCloneDirectory> localizedSites,
+            FailureCollector failureCollector) {
+        Map<Language, Boolean> localizedSitesUpToDate = new HashMap<>();
+        ObjectId latestMainRev = mainRepository.currentRev();
+        for (Map.Entry<Language, GitCloneDirectory> localized : localizedSites.entrySet()) {
+            localized.getValue().currentUpstreamRev().ifPresent(revId -> {
+                if (latestMainRev.equals(revId)) {
+                    localizedSitesUpToDate.put(localized.getKey(), true);
+                    // means that the localized repo points to the same rev that the main and we can ignore further checks
+                    return;
+                }
+                localizedSitesUpToDate.put(localized.getKey(), false);
+
+                // Otherwise let's see when the localized site was last synced,
+                // by looking at when that latest rev is pointing to:
+                Instant lastSync = GitUtils.revInstant(mainRepository.git(), revId);
+                // if the last sync of a localized site happened more than 7 days compared to the
+                Duration duration = Duration.between(lastSync, Instant.now());
+                if (duration.compareTo(Duration.ofDays(7)) > 0) {
+                    failureCollector.warning(
+                            FailureCollector.Stage.TRANSLATION,
+                            "Localized repository '" + localized.getKey() + "' is out of sync for " + duration);
+                }
```

Comment:
The concept of making a check makes sense, but here you're only checking the date on the main repository, so I'm not sure *that* makes sense. If the main repo gets pushed to every day, this check will never raise a warning.

Maybe you could check when the `upstream` submodule was last updated... somehow?